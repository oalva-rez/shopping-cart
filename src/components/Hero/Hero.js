import React from "react";
import womanHero from "../../assets/woman-hero.png";
import wallHero from "../../assets/wall-hero.png";

import { Link, useNavigate } from "react-router-dom";

export default function Hero() {
  const navigate = useNavigate();
  return (
    <section className="hero">
      <div className="hero-container">
        <img src={womanHero} alt="woman hero" />
        <div className="wall-hero">
          <h2>MUST HAVE BASICS</h2>
          <h3>
            Top quality clothing fabric brought to your doorstep shipped
            directly from the heart of Italy.
          </h3>
          <button onClick={() => navigate("/products")}>Shop Now</button>
        </div>
      </div>
    </section>
  );
}
