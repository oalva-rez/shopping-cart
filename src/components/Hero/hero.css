.hero-container {
  display: flex;
  width: 100%;
  height: clamp(60vh, 50vh, 50vh);
}
.hero-container > * {
  width: 50%;
  height: 100%;
  object-fit: cover;
}

.wall-hero {
  background-image: url("../../assets/wall-hero.png");
  display: grid;
  place-content: center;
}

.wall-hero > * {
  width: 80%;
  margin: auto;
  font-size: 1.2rem;
}
.wall-hero h2 {
  font-size: 2rem;
  margin-bottom: 7px;
  font-weight: bold;
}
.wall-hero h3 {
  font-weight: 500;
}
.wall-hero button {
  width: 200px;
  height: 30px;
  margin-top: 25px;
  font-weight: bold;
  font-size: 1.1rem;
  background-color: white;
  border: none;
  cursor: pointer;
}
