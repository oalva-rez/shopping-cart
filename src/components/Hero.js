import React from "react";
import hero from "../assets/hero.png";

export default function Hero() {
  return (
    <section className="hero">
      <img src={hero} alt="hero" />
      <h2>MUST HAVE BASICS</h2>
      <h3>
        Top quality clothing fabric brought to your doorstep shipped directly
        from the heart of Italy.
      </h3>
      <button>SHOP NOW</button>
    </section>
  );
}
