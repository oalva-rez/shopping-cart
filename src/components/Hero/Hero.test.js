import { render, screen } from "@testing-library/react";
import Hero from "./Hero";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  useNavigate: () => mockNavigate,
}));

describe("Hero", () => {
  it("matches snapshot", () => {
    const { container } = render(<Hero />);
    expect(container).toMatchSnapshot();
  });
  it('has link to products page "/products"', () => {
    render(<Hero />);
    const shopNowButton = screen.getByRole("button", { name: "Shop Now" });
    userEvent.click(shopNowButton);
    expect(mockNavigate).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenCalledWith("/products");
  });
});
