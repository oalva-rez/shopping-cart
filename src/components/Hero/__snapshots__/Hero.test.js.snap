// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero matches snapshot 1`] = `
<div>
  <section
    class="hero"
  >
    <div
      class="hero-container"
    >
      <img
        alt="woman hero"
        src="woman-hero.png"
      />
      <div
        class="wall-hero"
      >
        <h2>
          MUST HAVE BASICS
        </h2>
        <h3>
          Top quality clothing fabric brought to your doorstep shipped directly from the heart of Italy.
        </h3>
        <button>
          Shop Now
        </button>
      </div>
    </div>
  </section>
</div>
`;
